src/com/amazon/ion/benchmark/IonSchemaUtilities.java

```diff
@@ -101,25 +111,73 @@ public static int parseConstraints(IonStruct value, String keyWord) throws IOExc
+                    List<String> constraint = Arrays.asList(annotationValue.getTypeAnnotations());
+                    IonList annotations = (IonList) annotationValue;
+                    annotationList = checkOrdered(constraint, annotations, random);
+                }
+            }
+        }
+        return annotationList;
+    }
+
+    /**
+     * This is a helper method of getAnnotation which processes the constraint 'Ordered'.
+     * @param constraint is a List which contains all annotations of 'annotations' field.
+     * @param annotationList is the original annotation List without any consideration about constraints.
+     * @param random is a random integer generator.
+     * @return a List of processed annotations.
+     * @throws IOException if error occur when reading constraints.
+     */
+    private static IonList checkOrdered(List<String> constraint, IonList annotationList, Random random) throws IOException {
+        IonList result = annotationList;
+        if (!constraint.contains(IonSchemaUtilities.KEYWORD_ORDERED)) {
+            List<IonValue> annotations = annotationList.stream().collect(Collectors.toList());
+            Collections.shuffle(annotations);
+            try (IonReader shuffledAnnotationReader = IonReaderBuilder.standard().build(annotations.toString())) {
+                shuffledAnnotationReader.next();
+                result = (IonList) ReadGeneralConstraints.SYSTEM.newValue(shuffledAnnotationReader);
+            }
+        }
+        return checkRequired(constraint, result,random);
+    }
+
+    /**
+     * This is a helper method of getAnnotation which processes the constraint 'required'.
+     * @param constraint is a List which contains all annotations of 'annotations' field.
+     * @param annotationList is the annotation List after processing with the constraint 'ordered'.
+     * @param random is a random integer generator.
+     * @returna a List of processed annotations.
+     * @throws IOException if error occur when reading constraints.
+     */
+    private static IonList checkRequired(List<String> constraint, IonList annotationList, Random random) throws IOException {
+        IonList result = annotationList;
+        if (!constraint.contains(IonSchemaUtilities.KEYWORD_REQUIRED)) {
+            int randomValueOne = random.nextInt(annotationList.size());
+            int randomValueTwo = random.nextInt(annotationList.size());
+            List<IonValue> subAnnotationList = annotationList.subList(Math.min(randomValueOne, randomValueTwo), Math.max(randomValueOne, randomValueTwo));
+            if (subAnnotationList != null) {
+                try (IonReader annotationReader = IonReaderBuilder.standard().build(subAnnotationList.toString())) {
+                    annotationReader.next();
+                    result = (IonList) ReadGeneralConstraints.SYSTEM.newValue(annotationReader);
+                }
```

Comment:
Rather than writing and re-reading the list to get an `IonList`, just create an `IonList` and add the elements of `subAnnotationList` to it. See `IonSystem.newEmptyList`.